cmd/landscaper-controller/app/app.go

```diff
@@ -71,6 +77,84 @@ func (o *options) run(ctx context.Context) error {
 		return fmt.Errorf("unable to setup execution controller: %w", err)
 	}
 
+	///// WEBHOOK /////
+
+	// initialize webhook options
+	wo := webhook.Options{
+		WebhookConfigurationName: "landscaper-validation-webhook",
+		WebhookBasePath:          "/webhook/validate/",
+		WebhookNameSuffix:        ".validation.landscaper.gardener.cloud",
+		ObjectSelector: metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Operator: metav1.LabelSelectorOpNotIn,
+					Key:      "validation.landscaper.gardener.cloud/skip-validation",
+					Values:   []string{"true"},
+				},
+			},
+		},
+		ServicePort: o.webhookServicePort,
+	}
+	webhookLogger := ctrl.Log.WithName("webhook").WithName("validation")
+	webhookedResources := []webhook.WebhookedResourceDefinition{}
+	// add validation webhooks for Installations, DeployItems, and Executions (unless disabled)
+	if o.disabledWebhooks == nil || !o.disabledWebhooks["all"] {
+		if o.webhookNamespace == "" {
+			return errors.New("webhook service namespace must not be empty")
+		}
+		if o.webhookName == "" {
+			return errors.New("webhook service name must not be empty")
+		}
+		wo.ServiceNamespace = o.webhookNamespace
+		wo.ServiceName = o.webhookName
+
+		// generate certificates
+		mgr.GetWebhookServer().CertDir = filepath.Join(os.TempDir(), "k8s-webhook-server", "serving-certs")
+		wo.CABundle, err = webhook.GenerateCertificates(ctx, mgr, mgr.GetWebhookServer().CertDir, wo.ServiceNamespace, wo.ServiceName)
+		if err != nil {
+			return fmt.Errorf("unable to generate webhook certificates: %w", err)
+		}
+
+		// determine resources watched by webhooks
+		webhookedResourcesTemplate := []webhook.WebhookedResourceDefinition{
+			{
+				APIGroup:     "landscaper.gardener.cloud",
+				APIVersions:  []string{"v1alpha1"},
+				ResourceName: "installations",
+			},
+			{
+				APIGroup:     "landscaper.gardener.cloud",
+				APIVersions:  []string{"v1alpha1"},
+				ResourceName: "deployitems",
+			},
+			{
+				APIGroup:     "landscaper.gardener.cloud",
+				APIVersions:  []string{"v1alpha1"},
+				ResourceName: "executions",
+			},
+		}
+		webhookedResourcesLog := []string{}
+		for _, elem := range webhookedResourcesTemplate {
+			if !o.disabledWebhooks[elem.ResourceName] {
+				webhookedResources = append(webhookedResources, elem)
+				webhookedResourcesLog = append(webhookedResourcesLog, elem.ResourceName)
+			}
+		}
+		wo.WebhookedResources = webhookedResources
+		webhookLogger.Info("Enabling validation", "resources", webhookedResourcesLog)
+	}
+
+	// create/update/delete ValidatingWebhookConfiguration
+	if err := webhook.ApplyValidatingWebhookConfiguration(ctx, mgr, wo, webhookLogger); err != nil {
```

Comment:
should also be a noop if all webhooks are disabled

Reply:
No, if `all` is specified in the `disableWebhooks` option, this function will actively remove the `ValidatingWebhookConfiguration`, so it's not a noop. There is a corner-case, though: if `all` is not specified, but all webhooks are disabled (by specifying every single one of them instead of using `all`), this function will create/update an 'empty' ValidatingWebhookConfiguration. This is kind of a strange case - should we catch that and handle it similar to `all`? Probably a better idea than having it create a dummy webhook configuration ... (the `RegisterWebhooks` function already does nothing in this case, since it registers all non-deactivated webhooks, which causes it to be a noop if all have been deactivated)

Reply:
yes I would suggest to not deploy the webhook if no rules are defined.